core/src/main/java/com/google/adk/runner/Runner.java

```diff
@@ -366,8 +366,9 @@ private Single<Session> emitStateDeltaEvent(
   private InvocationContext newInvocationContextForLive(
       Session session, Optional<LiveRequestQueue> liveRequestQueue, RunConfig runConfig) {
     RunConfig.Builder runConfigBuilder = RunConfig.builder(runConfig);
-    if (!CollectionUtils.isNullOrEmpty(runConfig.responseModalities())
-        && liveRequestQueue.isPresent()) {
+    if (liveRequestQueue.isPresent() && !this.agent.subAgents().isEmpty()) {
+      // Parity with Python: apply modality defaults and transcription settings
+      // only for multi-agent live scenarios.
```

Comment:
@jinnigu The inline comment and the code don't seem to align, here? The "text" says "apply modality defaults" but then this removes `!CollectionUtils.isNullOrEmpty(runConfig.responseModalities()`... is that intentional? (It may well be, I'm entirely sure about why this was originally like this; but it seems worth double checking.) Also, why would we limit transcription only for multi-agent live scenarios? I would personally love to use this even for a very simple trivial only-LlmAgent use case... you speak to it, and get a persistent transcript in your session store, that's very cool! I'd love to use this e.g. in my (personal) https://docs.enola.dev project - but don't see why it needs to be limited to work only if `!this.agent.subAgents().isEmpty()`.

Reply:
The reason why I made the change like this is because I want to make adk-java equivalent to adk-python (https://github.com/google/adk-python/blob/main/src/google/adk/runners.py#L939-L971). I also agree that we should not limit transcription to multi-agent live scenarios only. I will raise an issue in adk-python to gather some feedbacks and make PRs to both adk-python and adk-java.